src/box/xrow.c

```diff
@@ -975,6 +985,18 @@ xrow_encode_dml(const struct request *request, struct region *region,
 		pos += tuple_len;
 		map_size++;
 	}
+	if (request->old_tuple != NULL) {
+		pos = mp_encode_uint(pos, IPROTO_OLD_TUPLE);
+		memcpy(pos, request->old_tuple, old_tuple_len);
+		pos += old_tuple_len;
+		map_size++;
+	}
+	if (request->new_tuple != NULL) {
+		pos = mp_encode_uint(pos, IPROTO_NEW_TUPLE);
+		memcpy(pos, request->new_tuple, new_tuple_len);
+		pos += new_tuple_len;
+		map_size++;
+	}
```

Comment:
Please add a test case to `test/unit/xrow.cc` that would check that this code works as expected.

Also, splitting key definition from the implementation makes the patch set more difficult to review. Please squash the first two patches.

Reply:
Ok, added test case for the whole `xrow_encode_dml()` function. Squashed patches as well.